utils/keys.go

```diff
@@ -0,0 +1,115 @@
+package utils
+
+import (
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// simple interface for the one function we need from the Storage interface
+type exportStore interface {
+	Get(string) ([]byte, error)
+	ListFiles() []string
+}
+
+type importStore interface {
+	Set(string, []byte) error
+}
+
+// ExportKeysByGUN exports all keys filtered to a GUN
+func ExportKeysByGUN(to io.Writer, s exportStore, gun string) error {
+	keys := s.ListFiles()
+	for _, k := range keys {
+		dir := filepath.Dir(k)
+		if dir == gun { // must be full GUN match
+			if err := ExportKeys(to, s, k); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportKeysByID exports all keys matching the given ID
+func ExportKeysByID(to io.Writer, s exportStore, ids []string) error {
+	want := make(map[string]struct{})
+	for _, id := range ids {
+		want[id] = struct{}{}
+	}
+	keys := s.ListFiles()
+	for _, k := range keys {
+		id := filepath.Base(k)
+		if _, ok := want[id]; ok {
+			if err := ExportKeys(to, s, k); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportKeys copies a key from the store to the io.Writer
+func ExportKeys(to io.Writer, s exportStore, from string) error {
+	// get PEM block
+	k, err := s.Get(from)
+	if err != nil {
+		return err
+	}
+
+	// parse PEM blocks if there are more than one
+	for block, rest := pem.Decode(k); block != nil; block, rest = pem.Decode(rest) {
+		// add from path in a header for later import
+		block.Headers["path"] = from
+		// write serialized PEM
+		err = pem.Encode(to, block)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ImportKeys expects an io.Reader containing one or more PEM blocks.
+// It reads PEM blocks one at a time until pem.Decode returns a nil
+// block.
+// Each block is written to the subpath indicated in the "path" PEM
+// header. If the file already exists, the file is truncated. Multiple
+// PEMs with the same "path" header are appended together.
```

Comment:
nitpick: Maybe "Multiple PEMs with the same "path" header that are adjacent to each other will be appended together?

Since if they are separated by PEM with a different path, the second one will overwrite the first.